_posts/2016-05-25-gwf-workshop.md

```diff
@@ -0,0 +1,994 @@
+* **`the_geom`** EPSG:4326. Unprojected coordinates in **decimal degrees** (Lon/Lat). WGS84 Spheroid.
+* **`the_geom_webmercator`** EPSG:3857. UTM projected coordinates in **meters**. This is a conventional Coordinate Reference System, widely accepted as a 'de facto' standard in webmapping. 
+
+In CartoDB, **the_geom_webmercator column is the one we see represented in the map**. Know more about projections:
+
+* In [this tutorial](http://docs.cartodb.com/tutorials/projections/).
+* [Map Projections in Wikipedia](https://en.wikipedia.org/wiki/Map_projection).
+* In [this CartoDB blog post](http://blog.cartodb.com/free-your-maps-web-mercator/).
+
+### 4.2 Changing map projections
+
+#### Accessing the **list of default projections** available in CartoDB:
+
+```sql
+SELECT 
+  * 
+FROM 
+  spatial_ref_sys
+```
+
+![srid](../img/160520-zgz/srid.png)
+
+#### Accessing the hidden **the_geom_webmercator** field:
+
+```sql
+SELECT 
+  the_geom_webmercator 
+FROM 
+  ne_50m_land
+```
+
+#### Adding **World Robinson** projection (ESPG:54030):
+
+```sql
+INSERT INTO spatial_ref_sys
+  (srid, auth_name, auth_srid, proj4text, srtext) values (54030, 'EPSG', 54030,
+'+proj=robin +datum=WGS84','PROJCS["World_Robinson",
+    GEOGCS["GCS_WGS_1984",
+        DATUM["WGS_1984",
+            SPHEROID["WGS_1984",6378137,298.257223563]],
+        PRIMEM["Greenwich",0],
+        UNIT["Degree",0.017453292519943295]],
+    PROJECTION["Robinson"],
+    PARAMETER["False_Easting",0],
+    PARAMETER["False_Northing",0],
+    PARAMETER["Central_Meridian",0],
+    UNIT["Meter",1],
+    AUTHORITY["EPSG","54030"]]');
+```
+
+```sql
+SELECT 
+  cartodb_id, ST_Transform(the_geom, 54030) AS the_geom_webmercator
+FROM 
+  ne_50m_land
+```
+
+![robinson](../img/160520-zgz/robinson.png)
+
+_About [`ST_Transform`](http://postgis.net/docs/ST_Transform.html)._
+
+### 4.3 Geoprocessing
+
+#### Create a **buffer** from points:
+
+```sql
+SELECT
+  cartodb_id,
+  name,
+  ST_Transform(
+    ST_Buffer(the_geom::geography, 50000)::geometry
+    ,3857
+  ) AS the_geom_webmercator
+FROM
+  populated_places
+WHERE
+  name ilike 'madrid'
+```
+
+![buffer](../img/160520-zgz/buffer.png)
+
+_About [`ST_Buffer`](http://postgis.net/docs/ST_Buffer.html)._
+
+#### Get the **difference** between two geometries:
+
+```sql
+SELECT
+  a.cartodb_id,
+    ST_Difference(
+        a.the_geom_webmercator,
+        b.the_geom_webmercator
+  ) AS the_geom_webmercator
+FROM
+  ne_50m_land a,
+  ne_adm0_europe b
+WHERE
+  b.adm0_a3 like 'ESP'
+```
+
+![difference](../img/160520-zgz/difference.png)
+
+_About [`ST_Difference`](http://postgis.net/docs/ST_Difference.html)._
+
+#### Get the number of points inside a polygon
+
+Using `GROUP BY`:
+
+```sql
+SELECT
+  b.cartodb_id,
+  b.name,
+  b.the_geom_webmercator,
+  count(*) AS pp_count,
+  sum(a.pop_max) as sum_pop
+FROM
+  ne_10m_populated_places_simple a,
+  ne_adm0_europe b
+WHERE
+  ST_Intersects(a.the_geom, b.the_geom)
+GROUP BY
+  b.cartodb_id,
+  b.name,
```

Comment:
as we discussed this morning, you probably can remove name and geom safely from the query